### PR TITLE
Update Compose file reference for version 3

### PR DESCRIPTION
**Pull request by aanand**
Currently a rough draft. This document will only be true once Engine 1.13 and Compose 1.10 are released.

TODO:

- [x] `healthcheck`
- [x] `stop_grace_period`

### CODE REVIEW

**Comment by johndmulhausen** [comment]
Deploy preview ready!

Built with commit 57b24497eba94a28d60e3ead8db267d2d0988222

https://deploy-preview-716--docker-docs.netlify.com

File: compose/compose-file.md [hunk 1]
```diff
@@ -847,10 +990,35 @@ easily retrieved and inspected using the docker command line or API.
...
+        image: db
+        volumes:
+          - data-volume:/var/lib/db
+      backup:
+        image: backup-service
+        volumes:
+          - data-volume:/var/lib/backup/data
+
+    volumes:
+      log-volume:
```
Comment (dnephin) [inline compose/compose-file.md]: `data-volume` I think

**Comment by aanand** [comment]
Added docs for `healthcheck` and `stop_grace_period`

File: compose/compose-file.md [hunk 2]
```diff
@@ -415,6 +532,38 @@ Running `id` inside the created container will show that the user belongs to
...
+Configure a check that's run to determine whether or not containers for this
+service are "healthy". See the docs for the
+[HEALTHCHECK Dockerfile instruction](/engine/reference/builder.md#healthcheck)
+for details on how healthchecks work.
+
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost"]
+      interval: 1m30s
+      timeout: 10s
+      retries: 3
```
Comment (dnephin) [inline compose/compose-file.md]: Don't we have `disable: true` as well?

File: compose/compose-file.md [hunk 3]
```diff
@@ -770,20 +932,39 @@ If you do not use a host path, you may specify a `volume_driver`.
...
+    [top-level `volumes` option](compose-file.md#volume-configuration-reference).
+
+-   For [version 1 files](compose-file.md#version-1), both named volumes and
+    container volumes will use the specified driver.
 
-> Note: No path expansion will be done if you have also specified a
-> `volume_driver`.
+-   No path expansion will be done if you have also specified a `volume_driver`.
+    For example, if you specify a mapping of `./foo:/data`, the `./foo` part
+    will be passed straight to the volume driver without being expanded.
```
Comment (dnephin) [inline compose/compose-file.md]: I didn't know this was a thing we did.

Comment (aanand) [inline compose/compose-file.md]: Yep, check out `test_home_directory_with_driver_does_not_expand` in `config_test.py`

File: compose/compose-file.md (hunk 2 above)
Comment (aanand) [inline compose/compose-file.md]: Good catch, I've added a note at the bottom and removed the `test: ["NONE"]` example.

**Comment by joaofnfernandes** [comment]
@mstanleyjones, @johndmulhausen this is an example of a file that deserves to have proper versioning. We should consider having a file for each compose file format. That way we can teach new users how to do things the new way so that they don't even learn the old format.

**Comment by londoncalling** [comment]
LGTM

**Comment by joaofnfernandes** [comment]
Isn't this supposed to be updated for 1.13? If so, shouldn't we merge this in `vnext-engine`, and not `master`?

**Comment by londoncalling** [comment]
@dnephin @aanand @nathanleclaire did you want this merged into `master` or `vnext`? I just got a ping from both Daniel and Nathen to merge this PR.

**Comment by johndmulhausen** [comment]
@aanand @londoncalling @dnephin @mstanleyjones 

I think we meant to do this in `vnext-compose` as it's for Compose v1.10.

It should also be tracked against the Compose v1.10 milestone: https://github.com/docker/docker.github.io/milestone/5

So, one more revert, and one more re-do. Sorry. 

CC'ing @mstanleyjones because I'm not sure when the last time we rebased `vnext-compose` was.

**Pull request merged**